### PR TITLE
Provide unit tests for SslConfigurationProvider.

### PR DESCRIPTION
**Pull request by mccheah**
Starting to unit test and harden everything we have so far.

### CODE REVIEW

**Comment by mccheah** [comment]
@kimoonkim @ssuchter for review

File: resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SslConfigurationProviderSuite.scala [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  private type SecretsHandler = MixedOperation[Secret, SecretList, DoneableSecret, SecretResource]
+
+  private val APP_ID = "app-id"
+  private val KEYSTORE_PASSWORD = "keystore"
+  private val KEY_PASSWORD = "key"
+  private val TRUSTSTORE_PASSWORD = "truststore"
+  private val IP_ADDRESS = "192.168.99.100"
+  private val SSL_SECRET_DIR = s"$DRIVER_CONTAINER_SECRETS_BASE_DIR/$APP_ID-ssl"
+
+  private val sslFolder = Files.createTempDirectory("ssl-configuration-provider-suite").toFile
```
Comment (mccheah) [inline resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SslConfigurationProviderSuite.scala:53]: I couldn't quite find an elegant Scalatest equivalent to JUnit's `TemporaryFolder` Rule. I wonder if there's a better option than manually setting up the directory. We do this in `KubernetesSuite` as well.

File: resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SSLUtils.scala [hunk 2]
```diff
@@ -61,7 +61,8 @@ private[spark] object SSLUtils {
     keyStore.load(null, null)
     keyStore.setKeyEntry("key", keyPair.getPrivate,
       keyPassword.toCharArray, Array(jcaCertificate))
-    val tempDir = Files.createTempDirectory("temp-ssl-stores").toFile()
+    val tempDir = Files.createTempDirectory("temp-ssl-stores").toFile
+    tempDir.deleteOnExit()
     tempDir.deleteOnExit()
```
Comment (kimoonkim) [inline resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SSLUtils.scala:66]: No need to call `deleteOnExit` twice at line 65 and 66?

File: resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SslConfigurationProviderSuite.scala (hunk 1 above)
Comment (kimoonkim) [reply resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/SslConfigurationProviderSuite.scala:53]: The current approach looks fine to me. I see many tests in `core` uses `Utils.createTempDir`, which is similar to this code:
```
  /**
   * Create a temporary directory inside the given parent directory. The directory will be
   * automatically deleted when the VM shuts down.
   */
  def createTempDir(
      root: String = System.getProperty("java.io.tmpdir"),
      namePrefix: String = "spark"): File = {
    val dir = createDirectory(root, namePrefix)
    ShutdownHookManager.registerShutdownDeleteDir(dir)
    dir
  }
```

Here's code snippet from [SparkContextSuite](https://github.com/apache-spark-on-k8s/spark/blob/k8s-support-alternate-incremental/core/src/test/scala/org/apache/spark/SparkContextSuite.scala):
```
  test("basic case for addFile and listFiles") {
    val dir = Utils.createTempDir()

    val file1 = File.createTempFile("someprefix1", "somesuffix1", dir)
    val absolutePath1 = file1.getAbsolutePath
```

Maybe we want to switch to this to be consistent. But, I don't have a strong preference here. i.e. I think it's your call.

**Review by kimoonkim — APPROVED** [review]
Looks great to me.

I don't have much experience with Scala unit test code, but this test code is very easy to read and understand. Thanks for writing this change. @mccheah.

I think it's good to go after addressing minor comments of mine below.

**Comment by mccheah** [comment]
Think I want #173 to merge first actually since that PR heavily refactors the class in question.

**Comment by ash211** [comment]
Please rebase onto `branch-2.1-kubernetes` and send this PR into that branch instead of `k8s-support-alternate-incremental` which is now deprecated.

**Comment by mccheah** [comment]
This is likely out of date with V2 submission work.

**Pull request closed without merging**